apigateway/internal/v1router/handler/payment_method.go

```diff
@@ -0,0 +1,71 @@
+package v1handler
+
+import (
+	"github.com/GrabItYourself/giys-backend/auth/pkg/authutils"
+	"github.com/GrabItYourself/giys-backend/payment/pkg/paymentproto"
+	"github.com/gofiber/fiber/v2"
+	"github.com/pkg/errors"
+)
+
+func (h *Handler) HandleGetMyPaymentMethods(c *fiber.Ctx) ([]*paymentproto.PaymentMethod, error) {
+	identity, ok := c.Locals(authutils.IdentityKey).(*authutils.Identity)
+	if !ok {
+		return nil, fiber.NewError(fiber.StatusUnauthorized, "identity not found in context")
+	}
+	ctx, err := authutils.EmbedIdentityToContext(c.Context(), identity)
+	if err != nil {
+		return nil, fiber.NewError(fiber.StatusInternalServerError, errors.Wrap(err, "can't embed identity to grpc context").Error())
+	}
+
+	res, err := h.Grpc.Payment.GetMyPaymentMethods(ctx, &paymentproto.GetMyPaymentMethodsRequest{})
+	if err != nil {
+		return nil, fiber.NewError(fiber.StatusInternalServerError, errors.Wrap(err, "Failed to request GRPC GetMyPaymentMethods").Error())
+	}
+
+	if len(res.PaymentMethods) == 0 {
+		return make([]*paymentproto.PaymentMethod, 0), nil
+	}
```

Comment:
why are we creating a new slice if len is 0? isn't this the exact same thing?

Reply:
empty slice is sent as nil in grpc.

changed to
```
if res.PaymentMethods == nil {
	return make([]*paymentproto.PaymentMethod, 0), nil
}
```